### DISCUSSION TITLE
Production setup: volumes

### DISCUSSION THREAD

**Discussion by Tanguille** [seaweedfs/seaweedfs#5069] [Q&A] [answered]
After experimenting with seaweedfs for a while I wanted to try to setup a production cluster by following `Production Setup` in the wiki. While reading it I came across a few questions: 
1. I have a test setup with 10 volumes running without specifying any `-max` argument. why wasn't I limited to 8 volume servers like described? I think this has something to do with me not understanding how volumes work properly. When reading at first it looks like you need one volume server per drive, while reading further it says "If the disk space is huge and there will be a lot of volumes" which makes me think huge drives do need multiple volume servers. What is considered huge, how many volume servers do I need per amount of GB's for example?
2. Up until now I never used the `-dir` argument but the guide tells you to use it. Why is that needed? What's the difference with not providing it?

I have read the entire wiki and searched for similar question but didn't come across any so I hope I didn't read over it. Thanks in advance.

**Comment by SmsS4** [comment, marked as answer]
If you are using docker `-max` is set to zero by default. look at [entrypoint.sh](https://github.com/seaweedfs/seaweedfs/blob/master/docker/entrypoint.sh)

**Reply by Tanguille** [reply]
I am using docker indeed. Thanks for clearing up some of my confusion.

**Reply by Tanguille** [reply]
So after checking out entrypoint.sh I understand better what is happening. It also does the `-dir` arugment for you which makes sense. I'm just wondering now how best to figure out how many volumes I need. Where can I find more information about that?

**Reply by SmsS4** [reply]
It depends on your use case
If you set your `volumeSizeLimitMB` too big each collection occupy big space in your disk and if you set `volumeSizeLimitMB` too small your number of volumes may be too much and this makes your write slow
You can always increase `volumeSizeLimitMB` and `-max`.

**Reply by Tanguille** [reply]
Okay thanks, I think I'll keep it as default for now and look for optimization if needed.
